src/headerssync.cpp

```diff
@@ -43,24 +43,24 @@ HeadersSyncState::HeadersSyncState(NodeId id, const Consensus::Params& consensus
     // could try again, if necessary, to sync a longer chain).
     m_max_commitments = 6*(Ticks<std::chrono::seconds>(NodeClock::now() - NodeSeconds{std::chrono::seconds{chain_start->GetMedianTimePast()}}) + MAX_FUTURE_BLOCK_TIME) / HEADER_COMMITMENT_PERIOD;
 
-    LogDebug(BCLog::NET, "Initial headers sync started with peer=%d: height=%i, max_commitments=%i, min_work=%s\n", m_id, m_current_height, m_max_commitments, m_minimum_required_work.ToString());
+    LogDebug(BCLog::NET, "Initial headers sync started with peer=%d: height=%i, max_commitments=%i, min_work=%s\n", m_id, m_presync_height, m_max_commitments, m_minimum_required_work.ToString());
 }
 
 /** Free any memory in use, and mark this object as no longer usable. This is
  * required to guarantee that we won't reuse this object with the same
  * SaltedTxidHasher for another sync. */
 void HeadersSyncState::Finalize()
 {
-    Assume(m_download_state != State::FINAL);
+    Assume(m_state != State::FINAL);
     ClearShrink(m_header_commitments);
-    m_last_header_received.SetNull();
+    m_presync_last_header_received.SetNull();
     ClearShrink(m_redownloaded_headers);
     m_redownload_buffer_last_hash.SetNull();
     m_redownload_buffer_first_prev_hash.SetNull();
     m_process_all_remaining_headers = false;
-    m_current_height = 0;
+    m_presync_height = 0;
 
-    m_download_state = State::FINAL;
+    m_state = State::FINAL;
```

Comment:
"state" seems a bit lazy to me, like naming a variable "thing" - it's a code smell hinting that we couldn't capture a well defined portion. Is this a state machine or why is it changing its internal state all the time? I know you didn't write it, but the rename doesn't help me in understanding it better.
The other renames are helpful.

Reply:
Better, thanks